clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -51,8 +51,6 @@
  * are held when they are invoked.
  */
 public class ConsumerNetworkClient implements Closeable {
-    private static final long MAX_POLL_TIMEOUT_MS = 5000L;
```

Comment:
Maybe we could restore this and do this in the constructor:
```
this.maxPollTimeoutMs = Math.min(maxPollTimeoutMs, MAX_POLL_TIMEOUT_MS)
```
That removes a bit of the awkwardness of using `5000` in `KafkaConsumer` without explanation and it ensures that there is no behavior change for the other usages of `ConsumerNetworkClient`.

Reply:
No problem, I will look into writing a test.